client/src/App.js

```diff
@@ -28,65 +32,118 @@ const globalStyles = makeStyles({
       }
     }
   }
-});
-
-const App = () => {
-  globalStyles();
-
-  const [toastProperties, setToastProperties] = useState({
-    text: '',
-    button: 'CLOSE',
-    variant: 'neutral'
-  });
-  const [showToast, toggleToast] = useState(false);
-  const [userAuthenticated, setAuthenticated] = useState(false);
-
-  const activateToast = (text, variant = 'neutral', button = 'CLOSE') => {
-    setToastProperties({ text, variant, button });
-    toggleToast(true);
+};
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    // globalStyles();
+
+    this.state = {
+      toastProperties: {
+        text: '',
+        button: 'CLOSE',
+        variant: 'neutral'
+      },
+      showToast: false,
+      userAuthenticated: false,
+      userDetails: {
+        name: '',
+        id: null
+      }
+    };
+
+    // Authenticate user pre-render
+    const jwtToken = localStorage.getItem('jwtToken');
+
+    if (jwtToken) {
+      try {
+        const data = jwtVerify(jwtToken, process.env.REACT_APP_JWT_SECRET).payload;
+        this.state.userDetails = { name: data.name, id: data._id };
+        this.state.userAuthenticated = true;
+      } catch (error) {
+        if (error.name === 'TokenExpiredError') {
+          localStorage.removeItem('jwtToken');
+          this.state = {
+            showToast: true,
+            toastDetails: {
+              text: 'Your session has expired, please log in again.',
+              variant: 'error',
+              ...this.state.toastDetails
+            },
+            ...this.state
+          };
+        } else {
+          console.log(error);
+        }
+      }
+    }
+  }
+
```

Comment:
I think this jwtToken checker could be moved out of App.js into a separate file as a helper function

Reply:
I did think about it, but I'd does relay on state, so I left it in there for now